batcher/src/lib.rs

```diff
@@ -81,13 +67,42 @@ impl App {
 
         Self {
             s3_client,
+            eth_rpc_provider,
             service_manager,
             sp1_prover_client,
+            eth_ws_url: config.eth_ws_url,
             current_batch: Mutex::new(Vec::new()),
+            block_interval: config.batcher.block_interval,
+            batch_size_interval: config.batcher.batch_size_interval,
+            last_uploaded_batch_block: Mutex::new(0),
         }
     }
 
-    pub async fn handle_connection(&self, raw_stream: TcpStream, addr: SocketAddr) {
+    pub async fn listen(self: &Arc<Self>, address: &str) {
+        // Create the event loop and TCP listener we'll accept connections on.
+        let listener = TcpListener::bind(address).await.expect("Failed to build");
+        // let listener = try_socket.expect("Failed to bind");
```

Comment:
remove this comment

Reply:
removed on 21dfe2ed26c7bad3231a2c9bb95b3f9203ac9b22